backend-plugin/src/main/java/com/redhat/jenkins/nodesharingbackend/Pool.java

```diff
@@ -284,4 +288,59 @@ public void generateResponse(StaplerRequest req, StaplerResponse rsp, Object nod
             rsp.sendError(HttpServletResponse.SC_NOT_IMPLEMENTED, getMessage());
         }
     }
+
+    // Helpers for JCasC
+    @CheckForNull
+    public String getConfigRepo() {
+        return System.getProperty(CONFIG_REPO_PROPERTY_NAME);
+    }
+
+    @DataBoundSetter
+    public void setConfigRepo(@Nonnull final String configRepo) {
+        String prop = getConfigRepo();
+        if (prop != null && prop.compareTo(configRepo) != 0) {
+            // Clashes
+            String msg = "Node-sharing Config Repo setting content clashes between environment ("
+                    + prop + ") and JCasC (" + configRepo + ")";
+            ADMIN_MONITOR.report(MONITOR_CONTEXT, new AbortException(msg));
+            throw new PoolMisconfigured(msg);
+        }
+        System.setProperty(CONFIG_REPO_PROPERTY_NAME, configRepo);
+    }
+
+    @CheckForNull
+    public String getUsername() {
+        return System.getProperty(USERNAME_PROPERTY_NAME);
+    }
+
+    @DataBoundSetter
+    public void setUsername(@Nonnull final String username) {
+        String prop = getUsername();
+        if (prop != null && prop.compareTo(username) != 0) {
+            // Clashes
+            String msg = "Node-sharing Config Repo setting content clashes between environment ("
+                    + prop + ") and JCasC (" + username + ")";
+            ADMIN_MONITOR.report(MONITOR_CONTEXT, new AbortException(msg));
+            throw new PoolMisconfigured(msg);
+        }
+        System.setProperty(USERNAME_PROPERTY_NAME, username);
+    }
+
+    @CheckForNull
+    public String getPassword() {
+        return System.getProperty(PASSWORD_PROPERTY_NAME);
+    }
+
+    @DataBoundSetter
+    public void setPassword(@Nonnull final String password) {
+        String prop = getPassword();
+        if (prop != null && prop.compareTo(password) != 0) {
+            // Clashes
+            String msg = "Node-sharing Config Repo setting content clashes between environment ("
+                    + prop + ") and JCasC (" + password + ")";
+            ADMIN_MONITOR.report(MONITOR_CONTEXT, new AbortException(msg));
```

Comment:
Not sure how we are handling it elsewhere, but might be worth not exposing the password values, just stating it differs.